src/omega_red/cache.clj

```diff
@@ -0,0 +1,33 @@
+(ns omega-red.cache
+  (:require
+    [omega-red.protocol :as redis]))
+
+
+(defprotocol Caching
+  (fetch-and-set
+    [this {:keys [fetch cache-set cache-get]}]
+    "Protocol for implementing read or fetch + cache workflow. At minium, `fetch-and-set` should return result of `fetch`"))
+
+
+(defn fetch-and-set* [redis {:keys [fetch ; function that fetches data
+                                    cache-set ; function that receives the fetch result and returns redis args for caching
+                                    cache-get ; vector of args to pass to redis, or a function that returns one to read from cache
+                                    ]}]
+  (or
+    ;; XXX: are read ops always returning falsey value for cache miss?
```

Comment:
`get` returns `nil` on miss but `smembers` return `()`

Reply:
Right, we could add a condition like this rather thån use `or`:

```clojure
(let [cache-read (cache-get)] 
  (if (or cache-read (seq cache-read))
    cache-read
    (do-the-fetch-and-set)))

```

or something like that